### PR TITLE
Accelerated pipelines examples for type adaptation

### PR DESCRIPTION
**Pull request by swapnesh-wani-nvidia**

### CODE REVIEW

File: rclcpp/type_adaptation/accelerated_pipeline/simple_increment/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,82 @@
+cmake_minimum_required(VERSION 3.5)
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/simple_increment/CMakeLists.txt:16]: Missing copyright header

File: rclcpp/type_adaptation/accelerated_pipeline/julia_set/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,142 @@
+cmake_minimum_required(VERSION 3.5)
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/julia_set/CMakeLists.txt:15]: Need copyright header

File: rclcpp/type_adaptation/accelerated_pipeline/julia_set/include/julia_set/cuda/julia_set.hpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/julia_set/include/julia_set/cuda/julia_set.hpp:2]: Need NVIDIA copyright line as well.

File: rclcpp/type_adaptation/accelerated_pipeline/julia_set/include/julia_set/julia_set_node.hpp [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  type_adaptation::example_type_adapters::ImageContainer,
+  sensor_msgs::msg::Image);
+
+namespace type_adaptation
+{
+namespace julia_set
+{
+
+/**
+ * @brief
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/julia_set/include/julia_set/julia_set_node.hpp]: Comment missing

File: rclcpp/type_adaptation/accelerated_pipeline/julia_set/src/cuda/julia_set.cu [hunk 5]
```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * NVIDIA CORPORATION and its licensors retain all intellectual property
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/julia_set/src/cuda/julia_set.cu]: Apache 2.0 header

File: rclcpp/type_adaptation/accelerated_pipeline/julia_set/package.xml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0"?>
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/julia_set/package.xml:1]: License / copyright header

File: rclcpp/type_adaptation/accelerated_pipeline/example_type_adapters/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,68 @@
+cmake_minimum_required(VERSION 3.5)
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/example_type_adapters/CMakeLists.txt:16]: Missing copyright header here

File: rclcpp/type_adaptation/accelerated_pipeline/example_type_adapters/src/image_container.cpp [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+  const sensor_msgs::msg::Image & sensor_msgs_image)
+: ImageContainer(std::make_unique<sensor_msgs::msg::Image>(sensor_msgs_image))
+{
+}
+
+ImageContainer::ImageContainer(const ImageContainer & other)
+{
+  nvtxRangePushA("ImageContainer:Copy");
+  header_ = other.header_;
+
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/example_type_adapters/src/image_container.cpp]: Spacing looks unnecessary

File: rclcpp/type_adaptation/accelerated_pipeline/example_type_adapters/src/image_container.cpp [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+  cuda_event_ = std::make_shared<CUDAEventWrapper>();
+  cuda_event_->record(other.cuda_stream_);
+  cudaStreamWaitEvent(cuda_stream_->stream(), cuda_event_->event());
+
+  // Copy the image data over so that this is an independent copy (deep copy).
+  cuda_mem_ = std::make_shared<CUDAMemoryWrapper>(size_in_bytes());
+  if (cudaMemcpyAsync(
+      other.cuda_mem_->device_memory(), cuda_mem_->device_memory(), size_in_bytes(),
+      cudaMemcpyDeviceToDevice, cuda_stream_->stream()) != cudaSuccess)
+  {
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/example_type_adapters/src/image_container.cpp:194]: Is this scope paren in the right place?

File: rclcpp/type_adaptation/accelerated_pipeline/example_type_adapters/src/image_container.cpp [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+    return *this;
+  }
+
+  header_ = other.header_;
+
+  height_ = other.height_;
+
+  width_ = other.width_;
+
+  encoding_ = other.encoding_;
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/example_type_adapters/src/image_container.cpp:211]: Unnecessary extra lines

File: rclcpp/type_adaptation/accelerated_pipeline/julia_set/launch/juliaset-pipeline-launch.py [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+from launch_ros.actions import ComposableNodeContainer
+from launch_ros.descriptions import ComposableNode
+
+RESOLUTIONS = {'16K': (15360, 8640),
+               '8K': (7680, 4320),
+               '4K': (3840, 2160),
+               '1080p': (1920, 1080),
+               '720p': (1280, 720),
+               '480p': (852, 480)}
+IMAGE_HZ = 100.0
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/julia_set/launch/juliaset-pipeline-launch.py]: Is this too fast?

File: rclcpp/type_adaptation/accelerated_pipeline/julia_set/launch/juliaset-pipeline-launch.py [hunk 12]
```diff
@@ -0,0 +1,139 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/julia_set/launch/juliaset-pipeline-launch.py]: julia_set-pipeline-launch.py

File: rclcpp/type_adaptation/accelerated_pipeline/simple_increment/src/inc_node.cpp [hunk 13]
```diff
@@ -0,0 +1,134 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/simple_increment/src/inc_node.cpp:2]: // Copyright 2022 NVIDIA CORPORATION

File: rclcpp/type_adaptation/accelerated_pipeline/julia_set/include/julia_set/julia_set_node.hpp [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+  const bool type_adaptation_enabled_;
+  // Current node number in the pipeline
+  const uint8_t proc_id_;
+  // Flag for intialization.
+  bool is_initialized;
+  // Counter
+  size_t counter_{0};
+  // Juliaset start x
+  float start_x_{0.0};
+  // Juliaset start y
```
Comment (hemalshahNV) [inline rclcpp/type_adaptation/accelerated_pipeline/julia_set/include/julia_set/julia_set_node.hpp]: JuliaSet?

File: rclcpp/type_adaptation/accelerated_pipeline/julia_set/launch/juliaset-pipeline-launch.py (hunk 11 above)
Comment (swapnesh-wani-nvidia) [reply rclcpp/type_adaptation/accelerated_pipeline/julia_set/launch/juliaset-pipeline-launch.py]: It works fine in testing.

**Review by hemalshahNV — APPROVED** [review]

**Review by hemalshahNV — APPROVED** [review]

**Pull request merged**